swayfmt/src/utils/map/comments.rs

```diff
@@ -193,9 +193,42 @@ fn get_comments_between_spans(
                 // There is a comment before this one, so we should get the context starting from the last comment's end to the beginning of the current comment
                 comments_with_context[index - 1].0.span.end()
             };
+
+            let mut rest_of_code = unformatted_code
+                .get(comment_span.end..)
+                .unwrap_or_default()
+                .lines()
+                .take(2);
+
+            // consume '\n'
+            let _ = rest_of_code.next();
+            // actual next line of code that we're interested in
+            let next_line = rest_of_code.next().unwrap_or_default();
+
+            // In certain cases where comments come in between unusual places,
+            // ..
+            //     }
+            //     // This is a comment
+            //     else {
+            // ..
+            // We need to know the context after the comment as well.
```

Comment:
It is great that this is working, really nice catch! My only slight-concern is that this a specialized solution, I cannot imagine any other scenario other than `else` that this stuff can happen again but it is possible. 

I guess without a major change of the way we are handling comments in the formatter this is the best we can do 👍

Reply:
Yeah, honestly I was (probably still am) concerned with such an approach, but seems like there's only 2 ways to handle formatting in our case:

1) make comments be aware of the next line of source code or 
2) format the source-code with it being aware of the positioning of comments.

Since we went with 1), I think for possible issues in future, we still might need to do some sort of 'look-ahead' to know if something needs to be indented.

In some [cases](https://github.com/FuelLabs/sway/pull/3556#issuecomment-1345820171) it seems to me that it might be better to have it be the other way around instead - e.g. if while formatting the source code, the formatter is aware of the comment width, we can choose to not inline the `if-else` block.

It made me think that at some point we might have to re-write at least part of the mechanism to make look-ahead possible while formatting the source code instead of the comments.

I rambled on a bit above but I think the immediate action here is probably to put this check in a function to let the next person working on this know that  this is something extendable for future lookaheads for other tokens.

Reply:
Yeah, as long as we are going with option 1, this feels inevitable. I think at some point we will want to have 2nd option. While trying to get the MVP for the formatter comments were a major problem we had to get done since there were no clear path to do that without introducing the way we are formatting. But I think if we can get a refactor to the formatter to insert comments **while** formatter visits each leaf node of the AST tree. This way we can also improve our comment handler greatly by simply inserting them alongside the items we are formatting. This would mean we would have comment-aware formatting like you mentioned in your comment. Also this could mean removing the extra parse step we are doing for comments and newlines. We have a related issue: https://github.com/FuelLabs/sway/issues/2357